dspace-spring-rest/src/main/java/org/dspace/app/rest/repository/RelationshipRestRepository.java

```diff
@@ -60,4 +94,100 @@ public RelationshipRest findOne(Context context, Integer integer) {
+
+        HttpServletRequest req = getRequestService().getCurrentRequest().getHttpServletRequest();
+        ObjectMapper mapper = new ObjectMapper();
+        RelationshipRest relationshipRest = null;
+        try {
+            relationshipRest = mapper.readValue(jsonNode.toString(), RelationshipRest.class);
+        } catch (IOException e) {
+            throw new UnprocessableEntityException("error parsing the body");
+        }
+        Relationship relationship = relationshipService.find(context, id);
+        if (relationship == null) {
+            throw new ResourceNotFoundException(apiCategory + "." + model + " with id: " + id + " not found");
+        }
+        Item leftItem = itemService.find(context, UUIDUtils.fromString(req.getParameter("leftItem")));
+        Item rightItem = itemService.find(context, UUIDUtils.fromString(req.getParameter("rightItem")));
+        RelationshipType relationshipType = relationshipTypeService
+            .find(context, Integer.parseInt(req.getParameter("relationshipType")));
+
+        if (authorizeService.authorizeActionBoolean(context, leftItem, Constants.WRITE) &&
+            authorizeService.authorizeActionBoolean(context, rightItem, Constants.WRITE)) {
+            relationship.setId(relationshipRest.getId());
+            relationship.setLeftItem(leftItem);
+            relationship.setRightItem(rightItem);
+
+            relationship.setRelationshipType(relationshipType);
+            if (relationshipRest.getLeftPlace() != -1) {
+                relationship.setLeftPlace(relationshipRest.getLeftPlace());
+            }
+            if (relationshipRest.getRightPlace() != -1) {
+                relationship.setRightPlace(relationshipRest.getRightPlace());
+            }
+
+            relationshipService.updatePlaceInRelationship(context, relationship);
+            relationshipService.update(context, relationship);
+
+            return relationshipConverter.fromModel(relationship);
+        } else {
+            throw new AccessDeniedException("You do not have write rights on this relationship's items");
+        }
+    }
+
+    @Override
+    protected void delete(Context context, Integer id) throws AuthorizeException {
+        Relationship relationship = null;
+        try {
+            relationship = relationshipService.find(context, id);
+            if (relationship != null) {
+                if (authorizeService.authorizeActionBoolean(context, relationship.getLeftItem(), Constants.WRITE) &&
+                    authorizeService.authorizeActionBoolean(context, relationship.getRightItem(), Constants.WRITE)) {
+                    relationshipService.delete(context, relationship);
+                } else {
+                    throw new AccessDeniedException("You do not have write rights on this relationship's items");
+                }
+            }
+        } catch (SQLException e) {
+            log.error(e.getMessage(), e);
```

Comment:
I'd prefer to log a more descriptive error (again `e.getMessage()` can be null).  Maybe this should be:
`log.error("Error deleting Relationship specified by ID:" + id, e);`

Reply:
This has been updated